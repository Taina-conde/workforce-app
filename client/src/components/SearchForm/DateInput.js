import { MuiPickersUtilsProvider } from "@material-ui/pickers";
import { Field } from "formik";
import { DatePicker } from "formik-material-ui-pickers";
import DateFnsUtils from "@date-io/date-fns";
const DateInput = () => {
  return (
    <MuiPickersUtilsProvider utils={DateFnsUtils}>
      <Field component={DatePicker} label="Data de cadastro" name="dataCad" />
    </MuiPickersUtilsProvider>
  );
};
export default DateInput;
