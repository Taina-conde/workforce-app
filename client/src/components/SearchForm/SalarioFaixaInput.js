import { Field } from 'formik';
import Slider from "@material-ui/core/Slider";
import Typography from "@material-ui/core/Typography";
const SalarioFaixaInput = () => {
  return (
    <>
      <Typography id="range-slider" gutterBottom>
        Faixa salarial
      </Typography>
      <Field name="salario">
        {({
          field, // { name, value, onChange, onBlur }
          form: { touched, errors }, // also values, setXXXX, handleXXXX, dirty, isValid, status, etc.
          meta,
        }) => {
          console.log("field", field);
          return (
            <Slider
              value={field.value}
              step = {1000}
              max = {10000}
              onChange={(e, value) =>
                field.onChange({
                  ...e,
                  target: {
                    ...e.target,
                    name: field.name,
                    value,
                  },
                })
              }
              valueLabelDisplay="on"
              aria-labelledby="range-slider"
              getAriaValueText={() => (`R$${field.value}`)}
            />
          );
        }}
      </Field>
    </>
  );
};
export default SalarioFaixaInput;
