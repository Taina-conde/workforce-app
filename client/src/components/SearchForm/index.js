import { Formik } from "formik";
import { setLocale } from "yup";
import * as yup from "yup";
import FormControl from "@material-ui/core/FormControl";
import Select from "@material-ui/core/Select";
import Button from "@material-ui/core/Button";
import InputLabel from "@material-ui/core/InputLabel";
import MenuItem from "@material-ui/core/MenuItem";
import { makeStyles } from "@material-ui/core/styles";
import { criteriosArr } from "../../helpers";
import NomeInput from "../shared/NomeInput";
import CpfInput from "../shared/CpfInput";
import CargoInput from "../shared/CargoInput";
import DateInput from "./DateInput";
import UfInput from "../shared/UfInput";
import SalarioFaixaInput from "./SalarioFaixaInput";
import StatusInput from "../shared/StatusInput";
import { getBy} from "../../api";
import {formatDate} from "../../helpers";

setLocale({
  number: {
    min: "Deve conter 11 números",
  },
});
const validationSchema = yup.object({
  nome: yup.string(),
  cpf: yup.number().positive().integer().min(11),
});
const useStyles = makeStyles((theme) => ({
  root: {
    "& .MuiTextField-root": {
      margin: theme.spacing(1),
      width: "50ch",
    },
  },
  formControl: {
    margin: theme.spacing(1),
    minWidth: 200,
  },
  btn: {
      margin: theme.spacing(2),
  },
  salario : {
      width: "100%",
  }
}));

const SearchForm = () => {
  const classes = useStyles();
  return (
    <Formik
      initialValues={{
        criterioBusca: "",
        nome: "",
        cpf: "",
        cargo: "",
        dataCad: formatDate(),
        ufNasc: "",
        salario: [0, 10000],
        status: "",
      }}
      validationSchema={validationSchema}
      onSubmit={(values) => {
        console.log(" values em submit", values);
        getBy(values.criterioBusca, values[values.criterioBusca])
      }}
    >
      {(props) => {
        console.log("cpf", props.values.cpf);
        return (
          <form onSubmit={props.handleSubmit} className={classes.root}>
            <FormControl className={classes.formControl}>
              <InputLabel id="criterioBusca">
                Buscar funcionários por{" "}
              </InputLabel>
              <Select
                labelId="criterioBusca"
                id="criterioBusca"
                name="criterioBusca"
                value={props.values.criterioBusca}
                onChange={props.handleChange}
              >
                {criteriosArr.map((item, index) => (
                  <MenuItem key={index} value={item.id}>
                    {item.criterio}
                  </MenuItem>
                ))}
              </Select>
            </FormControl>
            {props.values.criterioBusca === "nome" && (
              <FormControl className={classes.formControl}>
                <NomeInput />
              </FormControl>
            )}
            {props.values.criterioBusca === "cpf" && (
              <FormControl className={classes.formControl}>
                <CpfInput />
              </FormControl>
            )}
            {props.values.criterioBusca === "cargo" && (
              <FormControl className={classes.formControl}>
                <CargoInput />
              </FormControl>
            )}
            {props.values.criterioBusca === "dataCad" && (
              <FormControl className={classes.formControl}>
                <DateInput />
              </FormControl>
            )}
            {props.values.criterioBusca === "ufNasc" && (
              <FormControl className={classes.formControl}>
                <UfInput />
              </FormControl>
            )}
              {props.values.criterioBusca === "salario" && (
              <FormControl className={[classes.formControl, classes.salario]}>
                <SalarioFaixaInput/>
              </FormControl>
            )}
            {props.values.criterioBusca === "status" && (
              <FormControl className={classes.formControl}>
                <StatusInput />
              </FormControl>
            )}


            <Button className = {classes.btn} color="primary" variant="contained" type="submit">
              Pesquisar
            </Button>
          </form>
        );
      }}
    </Formik>
  );
};

export default SearchForm;
