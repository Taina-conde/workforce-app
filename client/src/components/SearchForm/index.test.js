import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import SearchForm from '.';

describe('SearchForm', () => {
    it('renders dropdown correctly', () => {
        const { container, getByTestId, getByText } = render(
            <SearchForm />
        );

        expect(container).toMatchSnapshot()

        userEvent.change(getByTestId('search-dropdown'))

        expect(container).toMatchSnapshot()
        
        expect(getByText('Nome')).toBe(undefined)
    })
    it('', () => {})
});
