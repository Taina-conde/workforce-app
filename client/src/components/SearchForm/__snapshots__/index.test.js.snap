// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders dropdown correctly 1`] = `
<div>
  <form
    class="makeStyles-root-1"
  >
    <div
      class="MuiFormControl-root makeStyles-formControl-2"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink="false"
        id="criterioBusca"
      >
        Buscar funcionários por
         
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        data-testid="search-dropdown"
      >
        <div
          aria-haspopup="listbox"
          aria-labelledby="criterioBusca criterioBusca"
          class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
          id="criterioBusca"
          role="button"
          tabindex="0"
        >
          <span>
            ​
          </span>
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput"
          name="criterioBusca"
          tabindex="-1"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
    
  </form>
</div>
`;

exports[`SearchForm renders dropdown correctly 2`] = `
<div>
  <form
    class="makeStyles-root-1"
  >
    <div
      class="MuiFormControl-root makeStyles-formControl-2"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink Mui-focused Mui-focused"
        data-shrink="true"
        id="criterioBusca"
      >
        Buscar funcionários por
         
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline Mui-focused Mui-focused MuiInputBase-formControl MuiInput-formControl"
        data-testid="search-dropdown"
      >
        <div
          aria-haspopup="listbox"
          aria-labelledby="criterioBusca criterioBusca"
          class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
          id="criterioBusca"
          role="button"
          tabindex="0"
        >
          <span>
            ​
          </span>
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput"
          name="criterioBusca"
          tabindex="-1"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
    
  </form>
</div>
`;
